src/kafka-util/src/admin.rs

```diff
@@ -103,51 +158,24 @@ where
                 .topics()
                 .iter()
                 .find(|t| t.name() == new_topic.name)
-                .ok_or(CreateTopicError::MissingMetadata)?;
+                .ok_or(anyhow!("unable to fetch topic metadata after creation"))?;
             // If the desired number of partitions is not "use the broker
             // default", wait for the topic to have the correct number of
             // partitions.
             if new_topic.num_partitions != -1 {
-                let num_partitions = i32::try_from(topic.partitions().len())
-                    .map_err(|_| CreateTopicError::TooManyPartitions)?;
-                if num_partitions != new_topic.num_partitions {
-                    return Err(CreateTopicError::PartitionCountMismatch {
-                        expected: new_topic.num_partitions,
-                        actual: num_partitions,
-                    });
+                let actual = i32::try_from(topic.partitions().len())?;
+                if actual != new_topic.num_partitions {
+                    bail!(
+                        "topic reports {actual} partitions, but expected {} partitions",
+                        new_topic.num_partitions
+                    );
                 }
             }
             Ok(false)
         })
         .await
 }
 
-/// An error while creating a Kafka topic.
```

Comment:
you inlined this one because each variant is only used in one place, and as it was before it was a  whole bunch of boilerplate?

Reply:
Brief rationale in https://github.com/MaterializeInc/materialize/pull/31869/commits/0297e073c3aebf51e56cc35c075dc5fe2ea11178

For me the main motive was... all the callers just convert to anyhow anyways. I had more logic (and more possible errors) to add, so it seemed like useless work to reshape the structure to add my own variants etc. when all that structure was immediately thrown away.